src/sidebar/components/Thread.js

```diff
@@ -129,4 +141,31 @@ function Thread({ thread, threadsService }) {
   );
 }
 
+/**
+ * Wrapper around AnnotationHeader
+ * @param {object} props
+ *   @param {Thread['annotation']} props.annotation
+ *   @param {number} props.replyCount
+ *   @param {boolean} props.threadIsCollapsed
+ */
+function AnnotationHeaderWrapper({ annotation, ...restProps }) {
+  const store = useStoreProxy();
+  const isSaving = annotation && store.isSavingAnnotation(annotation);
+  const isEditing = annotation && !!store.getDraft(annotation) && !isSaving;
```

Comment:
Copied from `Annotation` component.

Reply:
Any chance this logic could be hoisted up to `Thread` level and we can make `Annotation`, `AnnotationHeader` and this component a little "dumber" and DRY?